### DISCUSSION TITLE
Making a portal with a command block?

### DISCUSSION THREAD

**Discussion by carrotbane2** [iPortalTeam/ImmersivePortalsMod#1043] [Q&A]
Looking through the [Commands reference](https://github.com/qouteall/ImmersivePortalsMod/wiki/Commands-Reference), it seems like the vast majority are dependent on a player looking at a block. "create_small_inward_wrapping" can be done by a command block, so why can't I just make a single (biway, bifaced) portal at a specific location without needing to look at a block?

**Comment by qouteall** [comment]
all portal targeted commands can be used by command blocks by using `/execute as `  https://github.com/qouteall/ImmersivePortalsMod/wiki/Portal-Customization#edit-portals-using-command-blocks-and-functions

`portal cb_make_portal` can directly create a portal without the player.

**Reply by carrotbane2** [reply]
Okay, but can I make it biway, too?
```
/summon wither_skull <x> <y> <z> {Tags: [fromEntity]}
/summon wither_skull <x2> <y2> <z2> {Tags: [toEntity]}
/portal cb_make_poral @e[tag=fromEntity, limit=1] @e[tag=toEntity, limit=1]
/kill @e[tag=fromEntity]
/kill @e[tag=toEntity]
```
This makes a one-way, one-faced portal. To make it biway, I would need to access the portal entity I just created with a target selector. It would all be much simpler if it was possible to simply use the vanilla command, something like /summon immersiveportals:portal x y z <dimension> x2 y2 z2 {Rotation:[1.0, 0.0, 0.0]}.

**Reply by Classified3939** [reply]
While I agree that it could be made simpler, I use fromEntity as a marker to run portal set_portal_custom_name on the nearest portal (type=immersive_portals:portal,sort=nearest,limit=1), then use /execute as @e[name=customName] run portal complete_bi_way_portal.
So it isn't IMPOSSIBLE, but I agree the process is a little unwieldy.

**Reply by qouteall** [reply]
`/portal cb_make_portal` can have an extra argument that rename the portal now. for example `/portal cb_make_poral @e[tag=fromEntity, limit=1] @e[tag=toEntity, limit=1] "aaa"` then select that portal using `@e[name="xxx"]`

I forgot to mention it in the wiki. The wiki has updated.

**Reply by carrotbane2** [reply]
@Classified3939 Thanks, that works great. On a sidenote, do you know if there's a way to make portals inaccessible to all players? Simply giving it a fake playername (/portal set_portal_specific_accessor fakeName) doesn't work. Do I need to create some kind of dummy player entity?
@qouteall Doesn't work for me (immersive-portals-0.17-mc1.16.5-forge.jar)

**Reply by qouteall** [reply]
This feature was introduced in new Fabric versions. It's recommended to use the latest fabric version

**Reply by carrotbane2** [reply]
Okay... not gonna, though.
